test/e2e/components/vl-map.js

```diff
@@ -1,6 +1,94 @@
 const { VlElement } = require('vl-ui-core').Test;
+const { By } = require('vl-ui-core').Test.Setup;
+const VlMapBaseLayer = require('./vl-map-baselayer');
+const VlMapSearch = require('./vl-map-search');
+const VlMapOverviewMap = require('./vl-map-overview-map');
 
-class VlMap extends VlElement {  
+class VlMap extends VlElement {
+    async _getMap() {
+        return this.shadowRoot;
+    }
+
+    async getBaseLayers() {
+        const childElements = await this.findElements(By.css(':scope > *'));
+        const tagNames = await Promise.all(childElements.map(element => element.getTagName()));
+        const baseLayerElements = childElements.filter((element, index) => tagNames[index].startsWith('vl-map-baselayer'));
+        return Promise.all(baseLayerElements.map(element => new VlMapBaseLayer(this.driver, element)));
+    }
+
+    async isEscapeKeyDisabled() {
+        return this.hasAttribute("disable-escape-key");
+    }
+
+    async isRotationDisabled() {
+        return this.hasAttribute("disable-rotation");
+    }
+
+    async isMouseWheelZoomDisabled() {
+        return this.hasAttribute("disable-mouse-wheel-zoom");
+    }
+
+    async getOverviewMap() {
+        const map = await this._getMap();
+        await this.driver.wait(async () => {
+            const overviewMaps = await map.findElements(By.css('.ol-overviewmap'));
+            return overviewMaps.length > 0;
+        });
```

Comment:
is hier ook zo niks van Selenium met Expect visibility of ... ipv dat ge da zelf moet schrijven?

Reply:
await driver.wait(until.elementIsVisible(el),100);

Reply:
Dat is enkel als het element al bestaat, maar hier wordt dat nog asynchroon aangemaakt vermoed ik. elementLocated leek ook niet te werken vreemd goed.

Reply:
nee als t nie werkt ca va